openassessment/tests/test_data.py

```diff
@@ -152,3 +225,108 @@ def _load_fixture(self, fixture_relpath):
         )
         print "Loading database fixtures from {}".format(fixture_path)
         call_command('loaddata', fixture_path)
+
+
+class TestOraAggregateData(TransactionCacheResetTest):
+    """
+    Test that the component parts of OraAggregateData behave as expected.
+    """
+
+    def setUp(self):
+        super(TransactionCacheResetTest, self).setUp()
+        self.maxDiff = None
+        # Create submissions and assessments
+        self.submission = self._create_submission(STUDENT_ITEM)
+        self.scorer_submission = self._create_submission(SCORER_ITEM)
+        self.earned_points = 1
+        self.possible_points = 2
+        peer_api.get_submission_to_assess(self.scorer_submission['uuid'], 1)
+        self.assessment = self._create_assessment(self.scorer_submission['uuid'])
+
+        sub_api.set_score(self.submission['uuid'], self.earned_points, self.possible_points)
+        self.score = sub_api.get_score(STUDENT_ITEM)
+        peer_api.get_score(self.submission['uuid'], {'must_be_graded_by': 1, 'must_grade': 0})
+        self._create_assessment_feedback(self.submission['uuid'])
+
+    def _create_submission(self, student_item_dict):
+        """
+        Creates a submission and initializes a peer grading workflow.
+        """
+        submission = sub_api.create_submission(student_item_dict, ANSWER)
+        submission_uuid = submission['uuid']
+        peer_api.on_start(submission_uuid)
+        workflow_api.create_workflow(submission_uuid, STEPS)
+        return submission
+
+    def _create_assessment(self, submission_uuid):
+        """
+        Creates an assessment for the given submission.
+        """
+        return peer_api.create_assessment(
+            submission_uuid,
+            "scorer",
+            ASSESSMENT_DICT['options_selected'],
+            ASSESSMENT_DICT['criterion_feedback'],
+            ASSESSMENT_DICT['overall_feedback'],
+            RUBRIC_DICT,
+            2
+        )
+
+    def _create_assessment_feedback(self, submission_uuid):
+        """
+        Creates an assessment for the given submission.
+        """
+        feedback_dict = FEEDBACK_OPTIONS.copy()
+        feedback_dict['submission_uuid'] = submission_uuid
+        peer_api.set_assessment_feedback(feedback_dict)
+
+    def test_collect_ora2_data(self):
+        headers, data = OraAggregateData.collect_ora2_data(COURSE_ID)
+
+        self.assertEqual(headers, [
+            'Submission ID',
+            'Item ID',
+            'Anonymized Student ID',
+            'Date/Time Response Submitted',
+            'Response',
+            'Assessment Details',
+            'Assessment Scores',
+            'Date/Time Final Score Given',
+            'Final Score Points Earned',
+            'Final Score Points Possible',
+            'Feedback Statements Selected',
+            'Feedback on Peer Assessments'
+        ])
+
+        self.assertEqual(data[1], [
```

Comment:
Maybe I'm confused but why are we using `data[1]` here? Isn't `submission` created before `scorer_submission`? Or is the returned ordering not FIFO with respect to when the submissions were made?